subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/AnnotationProcessorDetector.java

```diff
@@ -117,13 +151,29 @@ public AnnotationProcessorDetector(FileContentCacheFactory cacheFactory) {
             }
         }
 
-        private List<AnnotationProcessorDeclaration> toProcessorDeclarations(List<String> processorNames) {
+        private Map<String, IncrementalAnnotationProcessorType> parseIncrementalProcessors(List<String> lines) {
```

Comment:
Would be nice to have a unit test for the incremental descriptor file parsing, at least to act as a spec.

Reply:
Done. And also simplified the other test which no longer needs to deal with detection, but just with the decision which path to use for processing.